std/path.d

```diff
@@ -3902,6 +3902,48 @@ if (isConvertibleToString!Range)
     assert(testAliasedString!isValidPath("/foo/bar"));
 }
 
+/** Checks whether `path1` is a subpath of `path2`
+  *
+  * Params:
+  *     path1 = string or Range of characters
+  *     path2 = string or Range of characters
+  *
+  * Returns:
+  *     true if `path` is a subpath of `path2`
+  *     false otherwise
+*/
+bool isSubPath(Range)(Range path1, Range path2)
+if ((isRandomAccessRange!Range && hasLength!Range && hasSlicing!Range &&
+    isSomeChar!(ElementEncodingType!Range) || isNarrowString!Range) &&
+    !isConvertibleToString!Range)
+{
+    import std.algorithm.searching : startsWith;
+    if (path1.startsWith("~")) path1 = expandTilde(path1);
+    if (path2.startsWith("~")) path2 = expandTilde(path2);
```

Comment:
None of the other functions in `std.path` do implicit tilde expansion, please remove